staging/src/k8s.io/kubectl/pkg/cmd/auth/cani.go

```diff
@@ -169,7 +169,12 @@ func (o *CanIOptions) Complete(f cmdutil.Factory, args []string) error {
 			if err != nil {
 				return err
 			}
-			o.Resource = o.resourceFor(restMapper, resourceTokens[0])
+			discoveryClient, err := f.ToDiscoveryClient()
```

Comment:
to have it in resourceFor. I guess we could do it in the beginning of the function - I just wanted to make the execution leaner in case we give an error to the user before that